### PR TITLE
docs: add fzf-lua and nu_plugin_skim to the README

### LINKED ISSUE #418: Add documentation for use with fzf-lua nvim plugin
Not really an issue, I saw that you are maintaining [`skim.vim`](https://github.com/lotabout/skim.vim), based off of `fzf.vim`.

I recently wrote [`fzf-lua`](https://github.com/ibhagwan/fzf-lua), as you can probably guess from the name it's an implementation of `fzf.vim` in pure lua, with many more features (icons, git indicators, native LSP support and more).

Yesterday I saw your project and due to my soft spot for any rust implementation of projects I love I decided to add support for `skim` (which is amazing btw), due to the great compatibility of the command line options it only took minor modifications to make it work and one slightly bigger modification to be able to use "interactive mode" for live grep (rg), which was utilizing the `change:reload` event in `fzf`.

If you'd like to try it out, follow the installation instructions on [my repo](https://github.com/ibhagwan/fzf-lua), ~~note that until [@vijaymarupudi](https://github.com/vijaymarupudi) decides what to do [with this PR](https://github.com/vijaymarupudi/nvim-fzf/pull/16) you need to use [my fork of nvim-fzf](https://github.com/ibhagwan/nvim-fzf)~~ and just change the `fzf_bin` variable to `sk` instead.

In short, add the below plugins (using `packer.nvim`):
```lua
use { 'ibhagwan/fzf-lua',
  requires = { 'nvim-tree/nvim-web-devicons' } -- optional for icons
}
```
or using `vim-plug`:
```vim
Plug 'ibhagwan/fzf-lua'
Plug 'nvim-tree/nvim-web-devicons'
```

And setup the plugin with skim:
```lua
lua << EOF
require'fzf-lua'.setup({fzf_bin = 'sk'})
EOF
```

Here's a glimpse of `fzf-lua` working with `skim` (left side live grep, right side LSP diagnostics):

screenshot-1627013466

### LINKED ISSUE #570: (not an actual issue) I'm making a Nushell plugin for integrating skim
I don't know if this project is still active, but I figured opening an issue would be the best way to notify the maintainers (if discussions were enabled here I'd open a discussion instead)

I'm working on a skim integration for [Nushell](http://www.nushell.sh/):

https://github.com/idanarye/nu_plugin_skim

Nushell is a shell that uses structured data types instead of raw text. It can still handle raw text, of course, but it provides its own versions of many shell commands that emit structured data. It also provides filters that can be used to shape that data.

My plugin uses skim as a library to create an `sk` Nushell command for filtering data. It's advantages over using regular `sk`:

1. You don't need to convert the structured Nushell streams to lines of regular text. You can just pipe them into `sk`.
2. The output is a structure Nushell object you can farther manipulate with other Nushell commands.
3. Some `sk` flags (like `--preview`) are Nushell closures that accepts the structured data as their input, instead of strings that take a column using a placeholder.

I'm hoping to make a release within a week or two (right now I'm working on supporting all the flags in `SkimOptions`). I'm giving you this heads up in case you want me to change anything before I make that 0.1.0 release.

### PR DESCRIPTION
**Pull request by LoricAndre**
## Checklist

_check the box if it is not applicable to your changes_
- [x] I have updated the README with the necessary documentation
- [x] I have added unit tests
- [x] I have added [end-to-end tests](test/test_skim.py)
- [x] I have linked all related issues or PRs
- [x] I have re-generated the completions and manpage using `cargo xtask compgen` and `cargo xtask mangen`

## Description of the changes

closes #570 
closes #418

### CODE REVIEW

**Comment by LoricAndre** [comment]
@idanarye, @ibhagwan could you please take a look and see if you're all right with this ?

File: README.md [hunk 1]
```diff
@@ -197,6 +198,32 @@ You can switch to `regex` mode dynamically by pressing `Ctrl-R` (Rotate Mode).
...
+{
+  "ibhagwan/fzf-lua",
+  -- optional for icon support
+  dependencies = { "nvim-tree/nvim-web-devicons" },
+  -- enable `sk` support instead of the default `fzf`
+  opts = {'skim'}
+}
+```
+
+## (wip) [nu_plugin_skim for nushell](https://github.com/idanarye/nu_plugin_skim)
```
Comment (idanarye) [inline README.md]: 1. I don't think it needs to be marked as `(wip)`. As far as I consider it's in a perfectly usable state.
2. Do you think there should also be a link here to [the Nushell website](https://www.nushell.sh/), or is it enough if users who don't know what Nushell is will click on the link to get to my repository and get to the Nushell website from there?

Comment (LoricAndre) [reply README.md]: Sorry, I was looking at the first issue you opened that mentioned it being pre-release without realizing that it was 2 years old 😅

**Review by idanarye — APPROVED** [review]

**Comment by ibhagwan** [comment]
```
  -- optional for icon support
  dependencies = { "nvim-tree/nvim-web-devicons" },
```

I'm not sure if we need this part, although nice to have there's also `mini.icons` (part of  the amazing [`mini.nvim`](https://github.com/echasnovski/mini.nvim)) which is also used by default with [`LazyVim`](https://github.com/LazyVim/LazyVim) - perhaps it's better to direct users to the repo's README for more options instead?

**Comment by LoricAndre** [comment]
I'll merge this tomorrow if it's good with you two

**Comment by ibhagwan** [comment]
> I'll merge this tomorrow if it's good with you two

Looks great, Ty @LoricAndre!

**Comment by jasha-hrp** [comment]
@LoricAndre it looks like these Readme updates got clobbered -- the README on main no longer has nushell plugin informatrion.

**Comment by LoricAndre** [comment]
Hmmm you're right, I need to check how that happened

**Pull request merged**